pkg/models/protocolvalidator.go

```diff
@@ -0,0 +1,16 @@
+//
+// Copyright (C) 2022 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package models
+
+import "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+
+// ProtocolValidator is a low-level device-specific interface implemented
+// by device services that validate device's protocol properties.
+type ProtocolValidator interface {
+	// Validate triggers device's protocol properties validation, returns error
+	// if validation failed and the incoming device will not be added into EdgeX.
+	Validate(protocols map[string]models.ProtocolProperties) error
```

Comment:
I suggest to rename this function to `ValidateProtocols` to prevent we have further validation requirements in the future.

Reply:
If we stay at the Device level on the interface the Validate should be fine. 
```suggestion
	Validate(device dtos.Device) error
```

Reply:
Should we use `models.Device` or `dtos.Device` here ?

Reply:
I guess models since it is internal and in the models package.

Reply:
@lenny-intel I still think this function should be `ValidateDevice` in case we want to support other object validation with a new interface.
Golang doesn't support different input parameters with the same function name (function overloading).

Reply:
@cloudxxx8,  Since is is on the DeviceValidator interface, we don't need to repeat 'Device' in the name of the API. That is call name stuttering and is to be avoided in Go.

Reply:
@lenny-intel for the protocolDriver implementation, it doesn't specify the interface name, but only function name.  For example, if we have a new ProfileAttributesValidator interface in the future, it can't use `Validate(attrs map[string]interface)` because the function name `Validate` is occupied.  We might want all the validator related function name explicit in the protocolDriver implementation, such as  `ValidateDevice`, `ValidateProvisionWatcher`, and `ValidateProfileAttributes`.

Reply:
I thought we chnaged `type ProtocolValidator interface` to `type DeviceValidator interface`

Reply:
but it can't validate the values in other objects, such as the attributes in ProvisionWatcher or any new object in the future.

Reply:
Correct, since the interface is specific to `DeviceValidator`. I don't think we want to add those validations to this interface. They would have their own optional interfaces to implement.

Reply:
Ok, I now get @cloudxxx8 point. I just looked at how this interface is used/implemented by the service (sample) and yes from that perspective the name needs to be `ValidateDevice`

Reply:
updated method name to be `ValidateDevice`